src/api/MarkerBasedAPI.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @flow
+ * @file class for Box marker based API's to inherit common functionality from
+ * @author Box
+ */
+import Base from './Base';
+
+type Params = {
+    marker: string,
+    limit: number,
+    params?: string
+};
+
+type Data = {
+    entries: Array<any>
+};
+
+class MarkerBasedApi extends Base {
+    /**
+     * @property {Data}
+     */
+    data: Data;
+
+    /**
+     * @property {string}
+     */
+    marker: string;
+
+    /**
+     * @property {number}
+     */
+    limit: number;
+
+    /**
+     * @property {Object}
+     */
+    params: ?Object;
+
+    /**
+     * @property {string}
+     */
+    id: string;
+
+    /**
+     * @property {boolean}
+     */
+    shouldFetchAll: boolean;
+
+    /**
+     * Gets query params for the API
+     *
+     * @return the query params object
+     */
+    getQueryParameters(marker: string, limit: number, params: ?Object): Object {
+        const queryParams: Params = {
+            marker,
+            limit
+        };
+
+        return Object.assign(queryParams, params);
+    }
+
+    /**
+     * Determines if the API has more items to fetch
+     * @param {string} marker the marker from the start to start fetching at
+     * @return {boolean} true if there are more items
+     */
+    hasMoreItems(marker: string): boolean {
+        return marker !== null && marker !== '';
+    }
+
+    /**
+     * Helper for get
+     *
+     * @private
+     */
+    async markerGetRequest(): Promise<void> {
+        if (this.isDestroyed()) {
+            return Promise.resolve();
+        }
+
+        // Make the XHR request
+        try {
+            const params = this.getQueryParameters(this.marker, this.limit, this.params);
+
+            const { data }: { data: Data } = await this.getData(this.id, params);
+
+            const entries = this.data ? this.data.entries : [];
+            this.data = {
+                ...data,
+                entries: entries.concat(data.entries)
+            };
+            this.marker = this.data ? this.data.next_marker : '';
+            const hasMoreItems = this.hasMoreItems(this.marker);
+            if (this.shouldFetchAll && hasMoreItems) {
+                return this.markerGetRequest();
+            }
+
+            this.successHandler(this.data);
+        } catch (error) {
+            this.errorHandler(error);
+        }
+
+        return Promise.resolve();
```

Comment:
No return needed either.
I see similar being done in OffSetBasedApi.
Can remove from there too.

Reply:
I dont think it can be removed, unless you want to suppress the eslint error

Reply:
You might be able to remove by changing ```return this.markerGetRequest();``` to ```this.markerGetRequest(); return;```